DOCUMENTATION.md

```diff
@@ -43,6 +45,153 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
+
+The other significant opportunity one leaves behind if you don't admit concurrency into your projections is that you can never split or load balance the processing across multiple consumers (Propulsion does not presently implement any explicit sharding support, although when using `Propulsion.CosmosStore`, the underlying Change Feed Processor in the Microsoft SDK implements automated balancing of physical partition leases across all competing instances).
+
+While doing the simplest thing possible to realise any aspect of a system's operation is absolutely the correct starting point for any design, it's also important to consider whether such solutions might impose restrictions that may later be difficult or costly to unravel:
+- re-traversing of all events to build a new version of a read model becomes less viable if the time to do that is hardwired to be limited by the serial processing of the items
+- while grouping multiple operations into a single SQL batch in order to reduce the latency (although not the locking side effects) can be effective, it should be noted that such a scheme does not generalise well to the e.g. third party API calls, or writes to document stores etc that do not present such facilities.
+- re-running processing in disaster recovery or amelioration scenarios will have restricted throughput (especially in cases where idempotent reprocessing might mainly involve relatively low cost operations; the lowered individual latency may be dwarfed by the per-call overheads in a way that the original serial processing of the requests might not have surfaced). It's easy to disregard such aspects as nice-to-haves when growing a system from scratch; the point is that there's nothing intrinsic about projection processing that mandates serial processing, and ruling out the opportunity for concurrent processing of streams can result in a system where powerful operational and deployment approaches are ruled out as a result of picking an Easy option over a more Simple one.
+- following the strict order of event writes across streams precludes grouping event processing at the stream level (unless you work in a batched manner). Being able to read ahead, collating events from future batches (while still honoring the batch order with regard to checkpointing etc) afford handlers the ability to process multiple events for a single stream as a group (equally, it allows a handler to determine that a given stream has already reached a particular write position, enabling the ingestion process to discard future redundant reads on the basis as processing catches up) 
+
```

Comment:
I wonder if this section needs to contain a caveat:

- Stream processing is not queue processing.
- If you need a queue, use a queue.
- Propulsion is not a replacement for the competing consumer pattern.
- There are limits to how far ahead you'd want to read without checkpointing. This can cause head-of-line blocking.

Reply:
About to add something addressing this - its definitely an important (and often misunderstood) point
Not understanding the head of line blocking point though

Reply:
If you have a poison message in a regular queue you quarantine it and move on (often by moving it to a dead letter queue). When processing a stream you block everything ahead of the poison message.

Reply:
ah, I had that covered in my text and was more than aware
was reading the wikipedia article and taking the networking meanign about reordering etc causing insurmountable backlogs (which can sometimes be a factor if you get batching and reading ahead wrong)
will ping when pushed

Reply:
Marking resolved. Further comments welcome on/at https://github.com/jet/propulsion/pull/200/commits/bec60b60e5aab861f01cc7059cc89d3caa7b1297